lib/crash_course_agent.ex

```diff
@@ -0,0 +1,35 @@
+defmodule CrashCourseAgent do
+  use GenServer
+  @moduledoc """
+  Gen Server with Agent API
+  """
+  # Client
+  def start_link(state \\ %{}) do
+    GenServer.start_link(__MODULE__, state, name: __MODULE__)
+  end
+
+  def get(), do: GenServer.call(__MODULE__, {:get})
+  def get(key), do: GenServer.call(__MODULE__, {:get, key})
+  def put(key, value), do: GenServer.call(__MODULE__, {:put, key, value})
+  def delete(key), do: GenServer.call(__MODULE__, {:delete, key})
+  def clear(), do: GenServer.call(__MODULE__, {:clear})
+
+  # Server (callbacks)
+  def init(state), do: {:ok, state}
```

Comment:
It’s a common practice to put `init` near `start_link`, since it is implicitly called by `start_link`.